### PR TITLE
Add base representation and update dataframe-related code

### PR DESCRIPTION
**Pull request by fzliu**
(Re-opening after rebase)
- Add a `BaseRepr` object from which all representation classes inherit
- Completed `DataframeRepr` based on example YAML, updated annotation-based initialization functions
- Updated dataframe-related code (still a WIP, needs further edits based on internal discussions)
- Updated docstrings and docstring formats for several DAG-related files
- Removed `DFIterator`, iterators can now be called directly from `Dataframe` objects
- Added test cases for `GraphRepr` and `DataframeRepr`
- Other slight formatting modifications

### CODE REVIEW

File: towhee/dag/dataframe_repr.py [hunk 1]
```diff
@@ -12,59 +12,82 @@
...
-    def __getitem__(self, key):
+            key:
+                Variable name.
+            value:
+                A pre-instantiated `VariableRepr` instance.
         """
-        Get a VariableRepr by its name
+        self._columns[key] = value
+
+    def from_input_annotations(self, func: FunctionType):
```
Comment (junjiejiangjjj) [inline towhee/dag/dataframe_repr.py:68]: In annotation, we should use "typing.Callable", not "types.FunctionType"

Comment (fzliu) [reply towhee/dag/dataframe_repr.py]: Fixed.

**Review by junjiejiangjjj — APPROVED** [review]
/approve
/lgtm

**Comment by fzliu** [comment]
#4

**Pull request merged**